packages/kbn-config/src/deprecation/unset_and_clean_empty_parent.test.ts

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { unsetAndCleanEmptyParent } from './unset_and_clean_empty_parent';
+
+describe('unsetAndcleanEmptyParent', () => {
+  test('unsets the property of the root object, and returns an empty root object', () => {
+    const config = { toRemove: 'toRemove' };
+    unsetAndCleanEmptyParent(config, 'toRemove');
+    expect(config).toEqual({});
+  });
+
+  test('unsets a nested property of the root object, and removes the empty parent property', () => {
+    const config = { nestedToRemove: { toRemove: 'toRemove' } };
+    unsetAndCleanEmptyParent(config, 'nestedToRemove.toRemove');
+    expect(config).toEqual({});
+  });
+});
```

Comment:
as a side effect of using `lodash.unset`, `unsetAndcleanEmptyParent` works for arrays as well. Let's either remove this possibility or document it via tests
```js
  test('does not remove a parent array', () => {
    const config = { nestedToRemove: [{ toRemove: 'toRemove' }] };
    unsetAndCleanEmptyParent(config, 'nestedToRemove[0].toRemove');
    expect(config).toStrictEqual({ nestedToRemove: [{}] });
  });
```

Reply:
Good point! I documented it with a comment and some tests.